components/org.wso2.carbon.identity.application.authenticator.facebook/src/main/java/org/wso2/carbon/identity/application/authenticator/facebook/FacebookAuthenticator.java

```diff
@@ -432,24 +475,57 @@ protected String getLoginType(HttpServletRequest request) {
         }
     }
 
-    @Override
-    public String getClaimDialectURI() {
-        String claimDialectUri = null;
+    private Map<String, String> readParametersFromAuthenticatorConfig() {
         AuthenticatorConfig authConfig = FileBasedConfigurationBuilder.getInstance().getAuthenticatorBean(getName());
         if (authConfig != null) {
-            Map<String, String> parameters = authConfig.getParameterMap();
-            if (parameters != null && parameters.containsKey(FacebookAuthenticatorConstants.
-                    CLAIM_DIALECT_URI_PARAMETER)) {
-                claimDialectUri = parameters.get(FacebookAuthenticatorConstants.CLAIM_DIALECT_URI_PARAMETER);
+            return authConfig.getParameterMap();
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("FileBasedConfigBuilder returned null AuthenticatorConfigs for the connector " +
+                        getName());
+            }
+            return null;
+        }
+    }
+
+    private ClaimConfig getAuthenticatorClaimConfigurations(AuthenticationContext context) {
+        ClaimConfig claimConfig = null;
+        if (context != null) {
+            ExternalIdPConfig externalIdPConfig = context.getExternalIdP();
+            if (externalIdPConfig != null) {
+                IdentityProvider identityProvider = externalIdPConfig.getIdentityProvider();
+                if (identityProvider != null) {
+                    claimConfig = identityProvider.getClaimConfig();
+                } else {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Authenticator " + getName() + " recieved null IdentityProvider");
+                    }
+                }
             } else {
                 if (log.isDebugEnabled()) {
-                    log.debug("Found no Parameter map for connector " + getName());
+                    log.debug("Authenticator " + getName() + " recieved null ExternalIdPConfig");
                 }
             }
         } else {
             if (log.isDebugEnabled()) {
-                log.debug("FileBasedConfigBuilder returned null AuthenticatorConfigs for the connector " +
-                        getName());
+                log.debug("Authenticator " + getName() + " recieved null AuthenticationContext");
+            }
+        }
+        return claimConfig;
+    }
+
+    @Override
+    public String getClaimDialectURI() {
+        String claimDialectUri = null;
```

Comment:
instead of null isn't it better to use StringUtils.EMPTY?

Reply:
When using empty, it breaks the flow from framework side.